### ISSUE TITLE
Discussion about the point tool

### ISSUE THREAD

**Issue by amanda-phet** [phetsims/graphing-quadratics#158]
Is this a common component? I can't find it in common code anywhere, and doesn't seem to be shared with graphing lines as far as I can tell. (I need a refresher, even though I'm sure we talked about this back when we worked on the sim). 

@Nancy-Salpepi brought up some cases where it might be nice to see the same value for the root and the value the point tool is detecting, instead of rounding on the point tool. 

![image](https://user-images.githubusercontent.com/7852916/129956108-0fd43459-2e8e-4f0d-aa64-e7cd7dd7c550.png)

The only way to make more space for it would be to make the grab area visually smaller. Is this feasible to do? Can the same design also go to other sims using the point tool, or is that not the case since it's not a common component?

**Comment by ariel-phet** [comment]
@amanda-phet I think you could certainly make the "three bar grab area" narrower horizontally.

1. If I recall, the reason for making the grab area this size would be to discourage occluding the readout with your finger while moving the tool. However, I am not sure that is of major concern considering the way this tool is used.

2. The pointer area for the tool is currently the entire tool (including the cross hairs), so I highly doubt reducing the size of the grab area would affect usability, being that the tool is reasonably large and obvious.

Basically it seems like a totally reasonable change (and my instincts would be it is not too cumbersome of a change to make)

**Comment by pixelzoom** [comment]
> Is this a common component? I can't find it in common code anywhere, and doesn't seem to be shared with graphing lines as far as I can tell. (I need a refresher, even though I'm sure we talked about this back when we worked on the sim).

Point tool is not common code. And there are in fact different implementation in graphing-quadratics and graphing-lines, because they have different requirements and work with different underyling models (straight lines vs quadratics).   I raised this concern in https://github.com/phetsims/graphing-quadratics/issues/18 during the implementation of graphing-lines. @amanda-phet and @ariel-phet participated in that discussion, and we decided to use different implementation for these 2 sims. 

graphing-quadratics and graphing lines DO share the same assert (.ai) file that was used to create the images.  From images/license.json:

```
"notes": "created by Bryce Gruneich, asset file is graphing-lines/assets/point_tool.ai"
```

> @Nancy-Salpepi brought up some cases where it might be nice to see the same value for the root and the value the point tool is detecting, instead of rounding on the point tool. The only way to make more space for it would be to make the grab area visually smaller. Is this feasible to do? 

I agree with @ariel-phet's answer to this question in https://github.com/phetsims/graphing-quadratics/issues/158#issuecomment-901377398.

Another option (which you've probably considered) is showing only 1 decimal place for roots and vertex. 

> Can the same design also go to other sims using the point tool, or is that not the case since it's not a common component?

Not the case. The only thing shared is the .ai file.  But if you're going to change graphing-quadratics, you should carefully consider what to do in graphing-lines.

The point tools in graphing-lines only display integer coordinates. So you'll be making the "display rectangle" much larger than is needed for graphing-lines.

If you don't mind the points tools having a different body size in graphing-quadratics and graphing-lines, then we should copy the .ai file to graphing-quadratics, modify that .ai file, generate new .png files, and make changes only to graphing-quadratics code.

If you want the point tools in graphing-quadratics and graphing-lines to have the same body size, then make the change to graphing-lines .ai file, general new .png files for both sims, and make code changes in both sims. 

Finally... If you don't mind ditching the "brushed metal" look of the tools, we could get rid of the .ai files and make these tools the appropriate size for the value that they need to display. Their look would be more similiar to the pseudo-3D probes used in various PhET sims. This will be a significant chunck of work, but is imo the best longterm approach.

**Comment by pixelzoom** [comment]
Here's an example of a 3D-looking tool in beers-law-lab.  It uses `ShadedRectangle`, which is in common code. 

<img width="231" alt="screenshot_1198" src="https://user-images.githubusercontent.com/3046552/130364256-87fe1cc1-39ee-4c04-bcde-b9adc0625d3e.png">

**Comment by amanda-phet** [comment]
Thanks for all of this background info, @pixelzoom 

I’d like to move forward with the improvements to the point tool. I really like the idea of the reusable component, and I prefer the appearance, but it would delay things significantly because Chris is going on vacation soon. Also, factoring out the point tool body later on sounds much easier than doing it now during the RC process. If getting the next RC out is the priority, then @pixelzoom and I think this is the best path forward: 

This week: change the body of the point tool in the ai file and make the “window” larger, which would be pretty quick to do (<1 hr)

After the PhET-iO publication: @pixelzoom will factor out the point tool body in common code, making a common point-tool-body-node in code that is more easily reusable (4-6 hours)

Mockup of potential new point tool on the right: 
![image](https://user-images.githubusercontent.com/7852916/130483864-1f956b86-a521-4e93-b2ed-28756dc51de7.png)

Waiting for @kathy-phet to confirm that we should move forward with this plan.

**Comment by kathy-phet** [comment]
I like the new mock-up of the "coded" point tool. 

Can you clarify will the point tool always be two decimals, or will it only go to two decimals on special points? 
If its always two decimal precision, will it be (9.1, 9.1) if it is (9.10, 9.10) point, or will (9,9) be displayed as (9.00, 9.00)? That is will it ever show only 1 decimal or no decimal places if it is precise value?

**Comment by pixelzoom** [comment]
> I like the new mock-up of the "coded" point tool.

To clarify... That is not what we're planning to do for 1.2 release.  That will be longterm work that will apply to graphing-quadratics, graphing-lines, and graphing-slope-intercept.

> Can you clarify will the point tool always be two decimals, or will it only go to two decimals on special points?

We are not proposing any change to current behavior. It will continue to work as it does now - trailing zeros are stripped. 

> Oh, and the plan of editting the .ai file now, and doing the body code later sounds OK. @pixelzoom - Do you anticipate a PhET-iO API change with that change?

No PhET-iO change should be necessary.

**Comment by kathy-phet** [comment]
Yes, I understand the body code would be later, and long term work. Thanks for clarifying on no API changes. If you are both happy to move forward, go for it!

**Comment by pixelzoom** [comment]
The long-term work to generalize "point tool body" and migrate it to common code is tracked in https://github.com/phetsims/graphing-quadratics/issues/167.

**Comment by pixelzoom** [comment]
Done, pushed to master and 1.2 branches.  @amanda-phet please review in master.

I asked @amanda-phet about whether to maintain a constant font size for the coordinate values, or to scale them to fit the point tool's display area.  The former would required a font size change, from 15 to 13.  She replied on Slack:
> I think the way it’s working now, shrinking to fit, is working fine, so that most of the time the coordinates can be larger.

**Comment by amanda-phet** [comment]
Something about the gradient was off, so I revised the artwork. 

Can you please integrate the newer artwork and assign back to me to check?

**Comment by pixelzoom** [comment]
Artwork revised in the above commits, in master and 1.2 branches. @amanda-phet please verify in master.

**Comment by amanda-phet** [comment]
Looks great- thank you!

**Comment by pixelzoom** [comment]
Ready for testing in the next RC.
